src/cache/http_cache.go

```diff
@@ -142,13 +142,14 @@ func (cache *httpCache) retrieve(target *core.BuildTarget, key []byte) (bool, er
 	resp, err := cache.client.Do(req)
 	if err != nil {
 		return false, err
-	} else if resp.StatusCode == http.StatusNotFound {
+	}
```

Comment:
This seems like a more interesting change. Maybe we should re-name the PR

Reply:
Done. This is ready for review now.